src/test/java/org/elasticsearch/common/xcontent/support/filtering/AbstractFilteringJsonGeneratorTests.java

```diff
@@ -0,0 +1,293 @@
+                                .endObject()
+                                .startObject()
+                                    .field("name", "Thetra")
+                                .endObject()
+                            .endArray()
+                        .endObject()
+                    .endObject()
+                .endObject()
+            .endObject();
+        return builder;
+    }
+
+    /**
+     * Instanciates a new XContentBuilder with the given filters and builds a sample with it.
+     */
+    private XContentBuilder sample(String... filters) throws IOException {
+        return sample(newXContentBuilder(filters));
+    }
+
+    @Test
+    public void testNoFiltering() throws Exception {
+        assertXContentBuilder(resultNoFiltering(), sample());
+        assertXContentBuilder(resultNoFiltering(), sample("*"));
+        assertXContentBuilder(resultNoFiltering(), sample("**"));
+    }
+
+    protected abstract String resultNoFiltering();
+
+    @Test
+    public void testNoMatch() throws Exception {
+        assertXContentBuilder(resultNoMatch(), sample("xyz"));
+    }
+
+    protected abstract String resultNoMatch();
+
+    @Test
+    public void testSimpleField() throws Exception {
+        assertXContentBuilder(resultSimpleField(), sample("title"));
+    }
+
+    protected abstract String resultSimpleField();
+
+    @Test
+    public void testSimpleFieldWithWildcard() throws Exception {
+        assertXContentBuilder(resultSimpleFieldWithWildcard(), sample("pr*"));
+    }
+
+    protected abstract String resultSimpleFieldWithWildcard();
+
+    @Test
+    public void testMultipleFields() throws Exception {
+        assertXContentBuilder(resultMultipleFields(), sample("title", "pages"));
+    }
+
+    protected abstract String resultMultipleFields();
+
+    @Test
+    public void testSimpleArray() throws Exception {
+        assertXContentBuilder(resultSimpleArray(), sample("tags"));
+    }
+
+    protected abstract String resultSimpleArray();
+
+    @Test
+    public void testSimpleArrayOfObjects() throws Exception {
+        assertXContentBuilder(resultSimpleArrayOfObjects(), sample("authors"));
+        assertXContentBuilder(resultSimpleArrayOfObjects(), sample("authors.*"));
+        assertXContentBuilder(resultSimpleArrayOfObjects(), sample("authors.*name"));
+    }
+
+    protected abstract String resultSimpleArrayOfObjects();
+
+    @Test
+    public void testSimpleArrayOfObjectsProperty() throws Exception {
+        assertXContentBuilder(resultSimpleArrayOfObjectsProperty(), sample("authors.lastname"));
+        assertXContentBuilder(resultSimpleArrayOfObjectsProperty(), sample("authors.l*"));
+    }
+
+    protected abstract String resultSimpleArrayOfObjectsProperty();
+
+    @Test
+    public void testRecurseField1() throws Exception {
+        assertXContentBuilder(resultRecurseField1(), sample("**.name"));
+    }
+
+    protected abstract String resultRecurseField1();
```

Comment:
You're not missing something, I did it in the `testRawField()` method but I can do it for all tests.